server/isomorphic-routing/index.js

```diff
@@ -37,6 +39,15 @@ export function serverRouter( expressApp, setUpRoute, section ) {
 	};
 }
 
+function setRouteMiddleware( context, next ) {
+	context.store.dispatch( setRouteAction(
+		context.pathname,
+		context.query
+	) );
+
+	next();
```

Comment:
Should work without `return`. @yurynix It's sufficient to call `next` to keep the chain intact. Normally, you'd use `return next()` to e.g. end execution of a middleware early (e.g. inside an `if` conditional) and move on to the next.